ui/pages/confirmations/components/advanced-gas-fee-popover/advanced-gas-fee-inputs/priority-fee-input/priority-fee-input.js

```diff
@@ -66,19 +66,20 @@ const PriorityFeeInput = () => {
     latestPriorityFeeRange,
     historicalPriorityFeeRange,
     priorityFeeTrend,
-  } = gasFeeEstimates;
+  } = gasFeeEstimates ?? {};
   const [priorityFeeError, setPriorityFeeError] = useState();
 
-  const [priorityFee, setPriorityFee] = useState(() => {
-    if (
-      estimateUsed !== PriorityLevels.custom &&
-      advancedGasFeeValues?.priorityFee &&
-      editGasMode !== EditGasModes.swaps
-    ) {
-      return advancedGasFeeValues.priorityFee;
-    }
-    return maxPriorityFeePerGas;
-  });
+  const defaultPriorityFee =
+    estimateUsed !== PriorityLevels.custom &&
+    advancedGasFeeValues?.priorityFee &&
+    editGasMode !== EditGasModes.swaps
+      ? advancedGasFeeValues.priorityFee
+      : maxPriorityFeePerGas;
+
+  const [priorityFee, setPriorityFee] = useState(defaultPriorityFee);
+  useEffect(() => {
+    setPriorityFee(defaultPriorityFee);
+  }, [defaultPriorityFee, setPriorityFee]);
```

Comment:
Here we did calculation in `useState` as this should be done only once. `defaultPriorityFee` should not change.

Reply:
I believe this is necessary because with the new polling hook some of the values from `useGasFeeContext` used to derive `defaultPriorityFee` will be undefined on first render. @jiexi can maybe add some color here

Reply:
But we need to check scenario if user has set a custom value for priority fee, can this override that ?

Reply:
User set Max Base Fee and Priority Fee are not reset/changed by this. I have verified by delaying the gas estimate state update in the hook and seeing that the Max Fee does not change and that of course the Max Base Fee and Priority Fee inputs do not change either